src/diffusers/pipelines/stable_cascade/pipeline_stable_cascade_prior.py

```diff
@@ -0,0 +1,629 @@
+            image_encoder=image_encoder,
+            feature_extractor=feature_extractor,
+            prior=prior,
+            scheduler=scheduler,
+        )
+        self.register_to_config(resolution_multiple=resolution_multiple)
+
+    # Copied from diffusers.pipelines.unclip.pipeline_unclip.UnCLIPPipeline.prepare_latents
+    def prepare_latents(self, shape, dtype, device, generator, latents, scheduler):
+        if latents is None:
+            latents = randn_tensor(shape, generator=generator, device=device, dtype=dtype)
+        else:
+            if latents.shape != shape:
+                raise ValueError(f"Unexpected latents shape, got {latents.shape}, expected {shape}")
+            latents = latents.to(device)
+
+        latents = latents * scheduler.init_noise_sigma
+        return latents
+
+    def encode_prompt(
+        self,
+        device,
+        batch_size,
+        num_images_per_prompt,
+        do_classifier_free_guidance,
+        prompt=None,
+        negative_prompt=None,
+        prompt_embeds: Optional[torch.FloatTensor] = None,
+        prompt_embeds_pooled: Optional[torch.FloatTensor] = None,
+        negative_prompt_embeds: Optional[torch.FloatTensor] = None,
+        negative_prompt_embeds_pooled: Optional[torch.FloatTensor] = None,
+    ):
+        if prompt_embeds is None:
+            # get prompt text embeddings
+            text_inputs = self.tokenizer(
+                prompt,
+                padding="max_length",
+                max_length=self.tokenizer.model_max_length,
+                truncation=True,
+                return_tensors="pt",
+            )
+            text_input_ids = text_inputs.input_ids
+            attention_mask = text_inputs.attention_mask
+
+            untruncated_ids = self.tokenizer(prompt, padding="longest", return_tensors="pt").input_ids
+
+            if untruncated_ids.shape[-1] >= text_input_ids.shape[-1] and not torch.equal(
+                text_input_ids, untruncated_ids
+            ):
+                removed_text = self.tokenizer.batch_decode(
+                    untruncated_ids[:, self.tokenizer.model_max_length - 1 : -1]
+                )
+                logger.warning(
+                    "The following part of your input was truncated because CLIP can only handle sequences up to"
+                    f" {self.tokenizer.model_max_length} tokens: {removed_text}"
+                )
+                text_input_ids = text_input_ids[:, : self.tokenizer.model_max_length]
+                attention_mask = attention_mask[:, : self.tokenizer.model_max_length]
+
+            text_encoder_output = self.text_encoder(
+                text_input_ids.to(device), attention_mask=attention_mask.to(device), output_hidden_states=True
+            )
+            prompt_embeds = text_encoder_output.hidden_states[-1]
+            if prompt_embeds_pooled is None:
+                prompt_embeds_pooled = text_encoder_output.text_embeds.unsqueeze(1)
```

Comment:
If it's a pooled embedding like SDXL, I think `pooled_prompt_embeds` makes more sense.

Reply:
@sayakpaul 
Umm, you're right. It makes sense, I think!
What do you think about `prompt_embeds` here? I just looked into SDXL I think its `prompt_embeds` is the hidden state, too (even though not the last one),  so the variable naming is consistent with SDXL here 

so I think it would be ok here but there is some inconsistency between the rest of pipelines here, no? I think the `prompt_embeds` is the pooled output for SD and most of our pipelines, no? Is there something we should clean up, maybe in a separate PR?

Reply:
Let me clear out the confusion. 

For most pipelines, `pooled_prompt_embeds` and `prompt_embeds` are not the same. 

This is because for most pipelines, we don't have `output_hidden_states` to True:

https://github.com/huggingface/diffusers/blob/4343ce2c8ef1593e6e630ddd77e792050973d9c4/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L403

But when it's True (e.g.,

https://github.com/huggingface/diffusers/blob/4343ce2c8ef1593e6e630ddd77e792050973d9c4/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L407

), the `prompt_embeds` is indexed differently (`prompt_embeds[-1]` as opposed to `prompt_embeds[0`).

This is quite different from how it's done in SDXL.

Let me know if you need more clarification.